modin/engines/base/io/text/text_file_reader.py

```diff
@@ -86,25 +86,24 @@ def offset(
             If file pointer reached the end of the file, but did not find
             closing quote returns `False`. `True` in any other case.
         """
-        general_kwargs = {
-            "nrows": 1,
-            "quotechar": quotechar,
-            "is_quoting": is_quoting,
-            "encoding": encoding,
-            "newline": newline,
-        }
-
         if is_quoting:
             chunk = f.read(offset_size)
-            outside_quotes = not chunk.count(quotechar) % 2
+            # outside_quotes = not chunk.count(quotechar) % 2
+            outside_quotes = (
+                not (chunk.count(quotechar) - chunk.count(b"\\" + quotechar)) % 2
```

Comment:
That is better but I'm not sure this handles cases like `\\"` (double slash meaning slash being itself escaped 😄 )... continue this up to 2N slashes or 2N+1 slashes followed by a quote 😉 

This also doesn't handle cases when a quote is not escaped by a slash (which might be possible - I don't know all csv dialects).

That said, I don't believe it's possible to 100% correctly parse the quotes without almost fully parsing the line, so we should probably leave this be as it is now.